index.d.ts

```diff
@@ -198,3 +198,36 @@ export type RequireAtLeastOne<ObjectType, KeysType extends keyof ObjectType = ke
 	}[KeysType]
 	// …then, make intersection types by adding the remaining properties to each mapped type.
 	& Omit<ObjectType, KeysType>;
+
+/*
+Create a type that requires at only one of the given properties. The remaining properties are kept as is.
+
+@example
+```
+import {RequireOnlyOne} from 'type-fest';
+
+type Responder = {
+	text: () => string;
+	json: () => string;
+
+	secure: boolean;
+};
+
+const responder: RequireOnlyOne<Responder, 'text' | 'json'> = {
+	json: () => '{"message": "ok"}',
+	secure: true
+};
+```
+*/
+export type RequireOnlyOne<ObjectType, KeysType extends keyof ObjectType = keyof ObjectType> =
```

Comment:
"Require only one" makes me think that it requires only one, but doesn't make it clear that it disallows more than one. Maybe we should simply name it `RequireOne`?